VogtObserver2/StockGrabber.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtObserver2
+{
+    class StockGrabber : IObservable
+    {
+        private HistoricalDataInfo _hdi = new HistoricalDataInfo();
+        private List<HistoricalDataResponse> _hdr;
+
+        private List<IObserver> _observers = new List<IObserver>();
+
+        public void AddObserver(IObserver o)
+        {
+            _observers.Add(o);
+        }
+
+        public void Notify()
+        {
+            foreach (IObserver observer in _observers)
+            {
+                observer.Update(GetPrice(), GetStockName()); 
+            }
+        }
+
+        public void RemoveObserver(IObserver o)
+        {
+            _observers.Remove(o);
+        }
+
+        public double GetPrice()
```

Comment:
This is the function that should be called from Program.cs. But I'd rename it so something like.. FetchLatestStocks. Then, for each of your "random" stocks, call thee API.

**this is the function that should be calling Notify() with the particular symbol that is being updated**